script.js

```diff
@@ -0,0 +1,442 @@
+const myobj={};
+
+function appendToInput(value) {
+    const textInput = document.getElementById('textInput');
+
+    const s = Array.from(textInput.value);
+
+    if(value !== "+" && value !== "-" && value !== "*" && value !== "/" && value !== "^" &&
+      value !== "sqrt(" && value !== "sin(" && value !== "cos(" && value !== "tan(" && value !== ")"  && value !== "pi"  && value !== "e")
+      {
+        const v = Number(value);
+        if(s[s.length-1] === ")")
+        {
+          textInput.value += value;
+          const w = Array.from(textInput.value);
+          w.splice(w.length-1,0,"*")
+          const result = w.join('');
+          textInput.value = result;
+          return;
+        }
+
+        if(s[s.length-1] === "i")
+        {
+          textInput.value += value;
+          const w = Array.from(textInput.value);
+          w.splice(w.length-1,0,"*")
+          const result = w.join('');
+          textInput.value = result;
+          return;
+        }
+
+        if(s[s.length-1] === "e")
+        {
+          textInput.value += value;
+          const w = Array.from(textInput.value);
+          w.splice(w.length-1,0,"*")
+          const result = w.join('');
+          textInput.value = result;
+          return;
+        }
+      }
+
+      if(value !== "+" && value !== "-" && value !== "*" && value !== "/" && value !== "^" &&
+      value === "sqrt(" || value === "sin(" || value === "cos(" || value === "tan("  || value === "pi"  || value === "e")
+      {
+        if(s[s.length-1] === "0" || s[s.length-1] === "1"  || s[s.length-1] === "2"  || s[s.length-1] === "3"  || s[s.length-1] === "4"  || s[s.length-1] === "5"  || s[s.length-1] === "6" || s[s.length-1] === "7" || s[s.length-1] === "8" || s[s.length-1] === "9" || s[s.length-1] === ")" ||s[s.length-1] === "i" || s[s.length-1] === "e" )
+        {
+          if(value == "sqrt(")
```

Comment:
change the if else to switch case, create a separate function for this functionality